engine/src/font/font.cpp

```diff
@@ -0,0 +1,55 @@
+#include <cubos/core/memory/stream.hpp>
+#include <cubos/core/reflection/external/primitives.hpp>
+#include <cubos/core/reflection/reflect.hpp>
+#include <cubos/core/reflection/traits/constructible.hpp>
+#include <cubos/core/reflection/type.hpp>
+
+#include <cubos/engine/font/font.hpp>
+
+CUBOS_REFLECT_IMPL(cubos::engine::Font)
+{
+    return cubos::core::reflection::Type::create("cubos::engine::Font");
+}
+
+namespace cubos::engine
+{
+    Font::Font(msdfgen::FreetypeHandle* ft, core::memory::Stream& stream)
+    {
+        if (ft == nullptr)
+        {
+            CUBOS_ERROR("Couldn't load font: Invalid Freetype handle");
+            return;
+        }
+        stream.seek(0, cubos::core::memory::SeekOrigin::End);
+        size_t streamSize = stream.tell();
+        auto* contents = static_cast<msdfgen::byte*>(operator new(sizeof(msdfgen::byte) * streamSize));
+        stream.seek(0, cubos::core::memory::SeekOrigin::Begin);
+        if (stream.read(contents, streamSize) < streamSize)
```

Comment:
If you change it to a vec
```suggestion
        if (stream.read(contents.data(), streamSize) < streamSize)
```